src/libs/actions/Report.js

```diff
@@ -649,6 +650,33 @@ function handleReportChanged(report) {
     reportMaxSequenceNumbers[report.reportID] = report.maxSequenceNumber;
 }
 
+/**
+ * Resets the Attachment Modal Data to default values.
+ */
+function clearAttachmentModalData() {
+    Onyx.set(ONYXKEYS.ATTACHMENT_MODAL, {
+        sourceURL: null, isAttachment: null, isModalOpen: false, file: null,
+    });
+}
+
+/**
+ * Sets the data for the attachment modal.
+ *
+ * @param {Object} reportID
+ * @param {String} reportID.sourceURL
+ * @param {Boolean} reportID.isAttachment
+ * @param {Boolean} reportID.isModalOpen
+ * @param {Object} reportID.file
+ * @param {String} reportID.file.name
+ */
+function setAttachmentModalData({
+    sourceURL, isAttachment, isModalOpen, file = null,
+}) {
+    Onyx.set(ONYXKEYS.ATTACHMENT_MODAL, {
+        sourceURL, isAttachment, isModalOpen, file,
+    });
+}
+
```

Comment:
Doesn't this only occur when the file is uploaded, therefore would only be shown in the ComposeAttachmentModal component?

Reply:
Also we're storing the state like this because the AttachmentModal's isModalOpen property needs to be controlled from a centralised position.

Reply:
Ok, so we need the `sourceURL`, `isAttachment`, and `isModalOpen`? Do we need `file`?

Reply:
I have no clue what the initial purpose of what the file attribute was in the codebase, I assume it was to show the file name for PDFs or something. But I simply kept it because I didn't think it was my place to be deleting functionality